limacharlie/retention.go

```diff
@@ -89,3 +89,102 @@ func (org *Organization) EventByAtom(sensorID, atom string) (EventContainer, err
 	err := org.client.reliableRequest(http.MethodGet, fmt.Sprintf("insight/%s/%s/%s", org.client.options.OID, sensorID, atom), q)
 	return event, err
 }
+
+type HistoricalDetectionsRequest struct {
+	// Cat is the category of the detections to fetch
+	Cat string `json:"cat"`
+	// Cursor is optional for paginated access, set to '-' for first query
+	Cursor string `json:"cursor"`
+	// Start is the required timestamp in seconds where to stop fetching detections
+	Start int `json:"start"`
+	// End is the required timestamp in seconds where to stop fetching detections
+	End int `json:"end"`
+	// Limit maximum number of detections to return
+	Limit int `json:"limit"`
+}
+
+func (org Organization) HistoricalDetections(detectionReq HistoricalDetectionsRequest) (HistoricalDetectionsResponse, error) {
+
+	var results HistoricalDetectionsResponse
+
+	if detectionReq.Cursor == "" {
+		detectionReq.Cursor = "-"
+	}
+
+	q := makeDefaultRequest(&results)
+	q = q.withQueryData(Dict{
+		"cat":    detectionReq.Cat,
+		"cursor": detectionReq.Cursor,
+		"start":  detectionReq.Start,
+		"end":    detectionReq.End,
+		"limit":  detectionReq.Limit,
+	})
+
+	if err := org.client.reliableRequest(http.MethodGet, fmt.Sprintf("insight/%s/detections", org.client.options.OID), q); err != nil {
+		return results, err
+	}
+
+	return results, nil
+}
+
+type HistoricalDetectionsResponse struct {
+	Detects    []Detects `json:"detects"`
+	NextCursor string    `json:"next_cursor"`
+}
+
+type Detect struct {
+	Event   Event   `json:"event"`
+	Routing Routing `json:"routing"`
+}
+type DetectMetadata struct {
+	Author         string   `json:"author"`
+	Description    string   `json:"description"`
+	Falsepositives []string `json:"falsepositives"`
+	Level          string   `json:"level"`
+	References     []string `json:"references"`
+	Tags           []string `json:"tags"`
+}
+
+type Detects struct {
+	Author    string         `json:"author"`
+	Cat       string         `json:"cat"`
+	Detect    Detect         `json:"detect"`
```

Comment:
Unfortunately we can't use a strict definition for the `detect` part. The definition you put down works well for some of the EDR data, but since LC supports all kinds of sources, the detect could contain Office365 or any other type of logs.
```suggestion
	Detect    Dict         `json:"detect"`
```